catroid/src/main/java/org/catrobat/catroid/utils/notifications/StatusBarNotificationManager.java

```diff
@@ -150,85 +135,65 @@ public int createDownloadNotification(Context context, String programName) {
 		PendingIntent pendingIntent = PendingIntent.getActivity(context, notificationId, downloadIntent,
 				PendingIntent.FLAG_CANCEL_CURRENT);
 
-		NotificationData data = new NotificationData(context, pendingIntent, R.drawable.ic_stat, programName,
-				R.string.notification_download_title_pending, R.string.notification_title_open,
-				R.string.notification_download_pending, R.string.notification_download_finished);
+		NotificationData data = new NotificationData(pendingIntent, R.drawable.ic_stat, programName,
+				context.getString(R.string.notification_download_title_pending), context.getString(R.string.notification_title_open),
+				context.getString(R.string.notification_download_pending), context.getString(R.string.notification_download_finished),
+				0, 100, false, false);
 
 		return createNotification(context, data);
 	}
 
 	private int createNotification(Context context, NotificationData data) {
 		initNotificationManager(context);
 
-		PendingIntent doesNothingPendingIntent = PendingIntent.getActivity(context, -1, new Intent(),
-				PendingIntent.FLAG_ONE_SHOT);
-
-		NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(context, CHANNEL_ID);
-		notificationBuilder.setContentTitle(data.getNotificationTitleWorking())
-			.setContentText(data.getNotificationTextWorking())
-			.setSmallIcon(data.getNotificationIcon())
-			.setOngoing(true)
-			.setContentIntent(doesNothingPendingIntent);
-
-		data.setNotificationBuilder(notificationBuilder);
+		data.setOngoing(true);
 		notificationDataMap.put(notificationId, data);
 
 		return notificationId++;
 	}
 
-	public Notification showOrUpdateNotification(int id, int progressInPercent) {
+	public void showOrUpdateNotification(int id, int progressInPercent) {
 		NotificationData notificationData = notificationDataMap.get(id);
 		if (notificationData == null) {
-			return null;
+			return;
```

Comment:
is it possible / wise to give some error information here or log something?

Reply:
logging would be ok, don't know about error information to the user since in this case clearly we messed up and we can't provide any useful information for a user. 

Since the NotificationManager should be refactored at some point, we'll probably restrain from holding all notification data in a map (which is just plain ugly) at some point in the hopefully near future

Reply:
Logging warning now